File: backend/app/schemas/reminder.py
import uuid
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional

class ReminderBase(BaseModel):
    days_before: int = Field(ge=0, description="Days before the due date. 0 means on the due date.")
    notification_enabled: bool = True
    alarm_enabled: bool = False
    voice_enabled: bool = False

class ReminderCreate(ReminderBase):
    pass

class ReminderUpdate(BaseModel):
    days_before: Optional[int] = Field(None, ge=0)
    notification_enabled: Optional[bool] = None
    alarm_enabled: Optional[bool] = None
    voice_enabled: Optional[bool] = None

class Reminder(ReminderBase):
    id: uuid.UUID
    financial_record_id: uuid.UUID
    user_id: uuid.UUID
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: flutter_app/lib/features/reminders/models/reminder.dart
class Reminder {
  final String id;
  final String financialRecordId;
  final int daysBefore;
  final bool notificationEnabled;
  final bool alarmEnabled;
  final bool voiceEnabled;

  Reminder({
    required this.id,
    required this.financialRecordId,
    required this.daysBefore,
    required this.notificationEnabled,
    required this.alarmEnabled,
    required this.voiceEnabled,
  });

  factory Reminder.fromJson(Map<String, dynamic> json) {
    return Reminder(
      id: json['id'],
      financialRecordId: json['financial_record_id'],
      daysBefore: json['days_before'],
      notificationEnabled: json['notification_enabled'],
      alarmEnabled: json['alarm_enabled'],
      voiceEnabled: json['voice_enabled'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'days_before': daysBefore,
      'notification_enabled': notificationEnabled,
      'alarm_enabled': alarmEnabled,
      'voice_enabled': voiceEnabled,
    };
  }
}
